File: WEEKLYCODE#1/left_rotation/Java/left_rotation.java

import java.util.*;
import java.lang.*;
import java.io.*;
import java.util.Scanner;

public class solution
{
	public static void main (String[] args) throws java.lang.Exception
	{
	    Scanner input = new Scanner(System.in);
		int n,d,i;
		//taking input 
		n=input.nextInt();
		d=input.nextInt();
		int []ar=new int[n];
		//taking input array
		for(i=0;i<n;i++)
		    ar[i]=input.nextInt();
		//printing array starting from d up to n    
		for(i=d;i<n;i++)
		    System.out.printf(ar[i]+" ");
		//printing array starting from 0 up to d
		for(i=0;i<d;i++)
		    System.out.printf(ar[i]+" ");
	}
}

File: WEEKLYCODE#1/left_rotation/C/left_rotation.c
#include <stdio.h>
#include <stdlib.h>
//main function
int main()
{
    int n,d,*ar,i;
    //declaring n,d for taking inputs where n:=size of the array & d:=number of rotations
    //declaring i as the loop variable
    //declaring a int pointer to make array
    scanf("%d%d",&n,&d);
    //allocating size to array , this is same as declaring an array of size n , ( int ar[n]; )
    ar=(int *)malloc(n*sizeof(int ));
    //taking the array as input
    for(i=0;i<n;i++)
        scanf("%d",&ar[i]);
    //printing array starting from d up to n
    for(i=d;i<n;i++)
        printf("%d ",ar[i]);
    //printing array starting from 0 up to d
    for(i=0;i<d;i++)
        printf("%d ",ar[i]);
    return 0;
}
